src/commands/scaling/addScaleRule/ScaleRuleNameStep.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ScaleRule } from '@azure/arm-appcontainers';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { localize } from '../../../utils/localize';
+import { IAddScaleRuleWizardContext } from './IAddScaleRuleWizardContext';
+
+export class ScaleRuleNameStep extends AzureWizardPromptStep<IAddScaleRuleWizardContext> {
+    public hideStepCount: boolean = true;
+    containerApp: ContainerApp | undefined;
+    scaleRules: ScaleRule[] | undefined;
+
+    public async prompt(context: IAddScaleRuleWizardContext): Promise<void> {
+        this.containerApp = context.containerApp.data;
+        this.scaleRules = context.scaleRuleGroup.data;
+        context.ruleName = (await context.ui.showInputBox({
+            prompt: localize('scaleRuleNamePrompt', 'Enter a name for the new scale rule.'),
+            validateInput: (value: string | undefined): string | undefined => this.validateInput(value)
+        })).trim();
+    }
+
+    public shouldPrompt(context: IAddScaleRuleWizardContext): boolean {
+        return context.ruleName === undefined;
+    }
+
+    private validateInput(name: string | undefined): string | undefined {
+        name = name ? name.trim() : '';
+
+        if (!/^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/.test(name)) {
+            return localize('invalidChar', `A name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character.`);
+        }
+
+        const scaleRuleExists: boolean = !!this.scaleRules?.some((rule) => {
+            return rule?.name?.length && rule?.name === name;
+        });
+        if (scaleRuleExists) {
+            return localize('scaleRuleExists', 'The scale rule "{0}" already exists in container app "{1}". Please enter a unique name.', name, this.containerApp?.name as string);
```

Comment:
nit: I don't think you need the `as string` typing at the end since localize doesn't care if it's undefined or not.